tests/tracer/test_processors.py

```diff
@@ -491,16 +509,16 @@ def test_single_span_sampling_processor_w_tracer_sampling(
 
 
 def test_single_span_sampling_processor_w_tracer_sampling_after_processing():
-    """Test that single span sampling tags and tracer sampling context are applied to spans
-    if the trace sampling is changed after the span is processed.
+    """Since the root span has MANUAL_KEEP_KEY set and the child span has not yet run through
```

Comment:
Previously, since the SpanSamplingProcessor ran on_span_finish, it would run, while  `root.context.sampling_priority == AUTO_REJECT`. Therefore causing span sampling to apply span sampling tags to the child span. Then doing  `root.set_tag(MANUAL_KEEP_KEY)` would cause the child span's context to update, leading to: `child.context.sampling_priority == USER_KEEP`. Therefore causing the child span to have both single span and trace sampling tags on it. 

For current behavior please read the doc string, I think this is actually better now.